src/client/java/net/neoforged/neoforge/client/CustomEnvironmentalEffectsRendererManager.java

```diff
@@ -0,0 +1,107 @@
+import org.jspecify.annotations.Nullable;
+
+/**
+ * Manager for custom renderers referred to by {@link net.minecraft.world.attribute.EnvironmentAttribute}.
+ */
+public final class CustomEnvironmentalEffectsRendererManager {
+    private static @Nullable Map<Identifier, CustomCloudsRenderer> CUSTOM_CLOUD_RENDERERS;
+    private static @Nullable Map<Identifier, CustomSkyboxRenderer> CUSTOM_SKYBOX_RENDERERS;
+    private static @Nullable Map<Identifier, CustomWeatherEffectRenderer> CUSTOM_WEATHER_EFFECT_RENDERERS;
+
+    /**
+     * Finds the {@link CustomCloudsRenderer} for a given identifier, or null if none is registered.
+     */
+    public static @Nullable CustomCloudsRenderer getCustomCloudsRenderer(Identifier id) {
+        if (NeoForgeEnvironmentAttributes.NO_CUSTOM_CLOUDS.equals(id)) {
+            return null;
+        }
+        return Objects.requireNonNull(CUSTOM_CLOUD_RENDERERS).get(id);
+    }
+
+    /**
+     * Finds the {@link CustomSkyboxRenderer} for a given identifier, or null if none is registered.
+     */
+    public static @Nullable CustomSkyboxRenderer getCustomSkyboxRenderer(Identifier id) {
+        if (NeoForgeEnvironmentAttributes.NO_CUSTOM_SKYBOX.equals(id)) {
+            return null;
+        }
+        return Objects.requireNonNull(CUSTOM_SKYBOX_RENDERERS).get(id);
+    }
+
+    /**
+     * Finds the {@link CustomWeatherEffectRenderer} for a given identifier, or null if none is registered.
+     */
+    public static @Nullable CustomWeatherEffectRenderer getCustomWeatherEffectRenderer(Identifier id) {
+        if (NeoForgeEnvironmentAttributes.NO_CUSTOM_WEATHER_EFFECTS.equals(id)) {
+            return null;
+        }
+        return Objects.requireNonNull(CUSTOM_WEATHER_EFFECT_RENDERERS).get(id);
+    }
+
+    /**
+     * Finds the {@link CustomWeatherEffectRenderer} to use for the given position in the given level.
+     */
+    public static @Nullable CustomWeatherEffectRenderer getCustomWeatherEffectRenderer(Level level, Vec3 position) {
+        var id = level.environmentAttributes().getValue(NeoForgeEnvironmentAttributes.CUSTOM_WEATHER_EFFECTS, position);
+        return getCustomWeatherEffectRenderer(id);
+    }
+
+    /**
+     * Finds the {@link CustomCloudsRenderer} to use for the given position in the given level.
+     */
+    public static @Nullable CustomCloudsRenderer getCustomCloudsRenderer(Level level, Vec3 position) {
+        var id = level.environmentAttributes().getValue(NeoForgeEnvironmentAttributes.CUSTOM_CLOUDS, position);
+        return getCustomCloudsRenderer(id);
+    }
+
+    /**
+     * Finds the {@link CustomSkyboxRenderer} to use for the given position in the given level.
+     */
+    public static @Nullable CustomSkyboxRenderer getCustomSkyboxRenderer(Level level, Vec3 position) {
+        var id = level.environmentAttributes().getValue(NeoForgeEnvironmentAttributes.CUSTOM_SKYBOX, position);
+        return getCustomSkyboxRenderer(id);
+    }
```

Comment:
These should be right next to their `Identifier`-accepting overload or at least in the same order as those overloads.